src/av2/evaluation/detection/eval.py

```diff
@@ -0,0 +1,246 @@
+    gts_uuids: List[str] = gts[uuid_column_names].to_numpy().astype(str).tolist()
+
+    # We merge the unique identifier -- the tuple of ("log_id", "timestamp_ns", "category")
+    # into a single string to optimize the subsequent grouping operation.
+    # `groupby_mapping` produces a mapping from the uuid to the group of detections / annotations
+    # which fall into that group.
+    uuid_to_dts = groupby([":".join(x) for x in dts_uuids], dts_npy)
+    uuid_to_gts = groupby([":".join(x) for x in gts_uuids], gts_npy)
+
+    log_id_to_avm: Optional[Dict[str, ArgoverseStaticMap]] = None
+    log_id_to_timestamped_poses: Optional[Dict[str, TimestampedCitySE3EgoPoses]] = None
+
+    # # Load maps and egoposes if roi-pruning is enabled.
+    if cfg.eval_only_roi_instances and cfg.dataset_dir is not None:
+        logger.info("Loading maps and egoposes ...")
+        log_ids = gts["log_id"].unique().tolist()
+        log_id_to_avm, log_id_to_timestamped_poses = load_mapped_avm_and_egoposes(log_ids, cfg.dataset_dir)
+
+    args_list: List[Tuple[NDArrayFloat, NDArrayFloat, DetectionCfg, Optional[ArgoverseStaticMap], Optional[SE3]]] = []
+    for uuid, sweep_gts in uuid_to_gts.items():
+        log_id, timestamp_ns, _ = uuid.split(":")
+        args: Tuple[NDArrayFloat, NDArrayFloat, DetectionCfg, Optional[ArgoverseStaticMap], Optional[SE3]]
+        args = uuid_to_dts[uuid], sweep_gts, cfg, None, None
+        if log_id_to_avm is not None and log_id_to_timestamped_poses is not None:
+            avm = log_id_to_avm[log_id]
+            city_SE3_ego = log_id_to_timestamped_poses[log_id][int(timestamp_ns)]
+            args = uuid_to_dts[uuid], sweep_gts, cfg, avm, city_SE3_ego
+        args_list.append(args)
+
+    logger.info("Starting evaluation ...")
+    outputs: Optional[List[Tuple[NDArrayFloat, NDArrayFloat]]] = Parallel(n_jobs=-1, verbose=1)(
+        delayed(accumulate)(*args) for args in args_list
+    )
+    if outputs is None:
+        raise RuntimeError("Accumulation has failed! Please check the integrity of your detections and annotations.")
+    dts_list, gts_list = zip(*outputs)
+
+    COLS = cfg.affinity_thresholds_m + TP_ERROR_COLUMNS + ("is_evaluated",)
+    dts_metrics: NDArrayFloat = np.concatenate(dts_list)  # type: ignore
+    gts_metrics: NDArrayFloat = np.concatenate(gts_list)  # type: ignore
+    dts.loc[:, COLS] = dts_metrics
+    gts.loc[:, COLS] = gts_metrics
+
+    # Compute summary metrics.
+    metrics = summarize_metrics(dts, gts, cfg)
+    metrics.loc["AVERAGE_METRICS"] = metrics.mean()
+    metrics = metrics.round(NUM_DECIMALS)
+    return dts, gts, metrics
+
+
+def summarize_metrics(
+    dts: pd.DataFrame,
+    gts: pd.DataFrame,
+    cfg: DetectionCfg,
+) -> pd.DataFrame:
+    """Calculate and print the 3D object detection metrics.
+
+    Args:
+        dts: (N,15) Table of detections.
+        gts: (M,15) Table of ground truth annotations.
+        cfg: Detection configuration.
```

Comment:
I'd add meaning for 15 here (e.g., where 15 is the number of attributes (e.g., x, y, z).